specs/Standards/Tokens/NonFungibleToken.md

```diff
@@ -0,0 +1,173 @@
+# Nonfungible Token
+
+## Summary
+
+[summary]: #summary
+
+A standard interface for non-fungible tokens allowing for ownership and transfer.
+
+## Motivation
+
+[motivation]: #motivation
+
+Non-fungible tokens (NFTs) have been described in many ways: digital goods, collectible items, unique online assets etc. The current core use case for NFTs is speculating on value and trading unique items. The use case of trading NFTs should be natively supported. This is the most basic set of functions needed to create an interoperable NFT that works in an asynchronous environment.
+
+Prior art:
+
+- [ERC-20 standard](https://eips.ethereum.org/EIPS/eip-20)
+- [ERC-721 standard](https://eips.ethereum.org/EIPS/eip-721)
+- [ERC-1155 standard](https://eips.ethereum.org/EIPS/eip-1155)
+
+## Guide-level explanation
+
+[guide-level-explanation]: #guide-level-explanation
+
+This token should allow the following:
+
+- Get the total supply of created tokens per contract.
+- Transfer a token to a new owner.
+- Grant access to all tokens to a third party.
+  - A third party account ID will be able to transfer the tokens that they have access to
+- Transfer tokens from one user to another.
+- Get current number of tokens for a given account ID. (Not in template).
+- Initialize contract once. The given total supply will be owned by the given account ID. (Not in template).
+
+There are a few concepts in the scenarios above:
+
+- **Total supply**. The total number of tokens in circulation.
+- **Token owner**. An account ID that owns one or more tokens.
+- **Transfer**. Action that moves some amount from one account to another account.
+- **Escrow**. A different account from the balance owner who has permission to one or more tokens.
```

Comment:
Let's call it an **Operator**. It's cleary indicating the use of that role: operating someones else token, (rather than holding it).

Reply:
Trying to keep consistent with ERC20. I think that `Escrow` is more accurate since they actually take ownership of the tokens in the current model.

Reply:
Escrow is a very specific process. Her we have more general functionality. According to [Investopedia](https://www.investopedia.com/terms/e/escrow.asp), escrow:
> Escrow is a legal concept describing a financial instrument whereby an asset or escrow money is held by a third party on behalf of two other parties that are in the process of completing a transaction. 
> An escrow agent is an entity that has fiduciary responsibilities in the transfer of property from one party to another. 

The differences here are:
1. It's an interface, and it's up to the implementation if the asset is changing an owner or not. In most of the implementations, we don't change an ownership - instead an allowance record is created. BTW: also you description doesn't say that we want to change the owner: _A different account from the balance owner who has permission to one or more tokens._
2. IMHO it's better to keep it general for different processes.